### PR TITLE
303 - Activity page - General Stationary Combustion (non-compression and non-processing)

### PR DESCRIPTION
**Pull request by pbastia**

### CODE REVIEW

File: bc_obps/reporting/json_schemas/2024/gsc_non_compression_non_combustion/with_useful_energy.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
+{
+  "type": "object",
+  "title": "General stationary combustion of fuel or waste with production of useful energy",
```
Comment (dleard) [inline bc_obps/reporting/json_schemas/2024/gsc_non_compression_non_combustion/with_useful_energy.json:3]:
```suggestion
  "title": "General stationary combustion of fuel or waste at a linear facilities operation resulting in the production of useful energy",
```
The LFO GSC activities have similar, but different source types

File: bc_obps/reporting/json_schemas/2024/gsc_non_compression_non_combustion/without_useful_energy.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
+{
+  "type": "object",
+  "title": "General stationary combustion of waste without production of useful energy",
```
Comment (dleard) [inline bc_obps/reporting/json_schemas/2024/gsc_non_compression_non_combustion/without_useful_energy.json:3]:
```suggestion
  "title": "General stationary combustion of fuel or waste at a linear facilities operation not resulting in the production of useful energy",
```
same as above

File: bc_obps/reporting/migrations/0007_reinit.py [hunk 3]
```diff
@@ -840,6 +840,12 @@ class Migration(migrations.Migration):
             ),
             preserve_default=False,
         ),
+        migrations.AddConstraint(
```
Comment (dleard) [inline bc_obps/reporting/migrations/0007_reinit.py:843]: :+1:

File: bc_obps/reporting/tests/models/program_configuration_tests/test_gsc_non_compression_non_processing_2024.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                'Heat Input/Default EF': 2,
+                'Alternative Parameter Measurement': 1,
+                'Replacement Methodology': 1,
+            },
+        }
+
+        config = {
+            'Field gas or process vent gas combustion at a linear facilities operation': gas_config,
+            'General stationary combustion of fuel or waste with production of useful energy': gas_config,
+            'General stationary combustion of waste without production of useful energy': gas_config,
```
Comment (dleard) [inline bc_obps/reporting/tests/models/program_configuration_tests/test_gsc_non_compression_non_processing_2024.py:55]:
```suggestion
            'General stationary combustion of fuel or waste at a linear facilities operation not resulting in the production of useful energy': gas_config,
```

File: bc_obps/reporting/tests/models/program_configuration_tests/test_gsc_non_compression_non_processing_2024.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                'Measured Steam/Default EF': 3,
+                'Heat Input/Default EF': 2,
+                'Alternative Parameter Measurement': 1,
+                'Replacement Methodology': 1,
+            },
+        }
+
+        config = {
+            'Field gas or process vent gas combustion at a linear facilities operation': gas_config,
+            'General stationary combustion of fuel or waste with production of useful energy': gas_config,
```
Comment (dleard) [inline bc_obps/reporting/tests/models/program_configuration_tests/test_gsc_non_compression_non_processing_2024.py:54]:
```suggestion
            'General stationary combustion of fuel or waste at a linear facilities operation resulting in the production of useful energy': gas_config,
```

File: bciers/apps/reporting/src/app/components/activities/uiSchemas/gscNonCompressionNonProcessingUiSchema.ts [hunk 6]
```diff
@@ -0,0 +1,228 @@
+import FieldTemplate from "@bciers/components/form/fields/FieldTemplate";
+import NestedArrayFieldTemplate from "@bciers/components/form/fields/NestedArrayFieldTemplate";
+import SourceTypeBoxTemplate from "@bciers/components/form/fields/SourceTypeBoxTemplate";
+import CheckboxWidgetLeft from "@bciers/components/form/widgets/CheckboxWidgetLeft";
+import InlineFieldTemplate from "@bciers/components/form/fields/InlineFieldTemplate";
+
+const uiSchema = {
+  "ui:FieldTemplate": FieldTemplate,
+  "ui:classNames": "form-heading-label",
+  gscWithProductionOfUsefulEnergy: {
```
Comment (dleard) [inline bciers/apps/reporting/src/app/components/activities/uiSchemas/gscNonCompressionNonProcessingUiSchema.ts:10]:
```suggestion
  gscWithProductionOfUsefulEnergy: {
```
gscFuelOrWasteLinearFacilitiesUsefulEnergy

File: bciers/apps/reporting/src/app/components/activities/uiSchemas/gscNonCompressionNonProcessingUiSchema.ts [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+    "ui:FieldTemplate": FieldTemplate,
+    "ui:widget": CheckboxWidgetLeft,
+    "ui:options": {
+      label: false,
+    },
+  },
+  gscWithoutProductionOfUsefulEnergy: {
```
Comment (dleard) [inline bciers/apps/reporting/src/app/components/activities/uiSchemas/gscNonCompressionNonProcessingUiSchema.ts:17]:
```suggestion
  gscFuelOrWasteLinearFacilitiesWithoutUsefulEnergy: {
```

File: bciers/apps/reporting/src/app/components/activities/uiSchemas/gscNonCompressionNonProcessingUiSchema.ts [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+    "ui:options": {
+      label: false,
+    },
+  },
+  sourceTypes: {
+    "ui:FieldTemplate": FieldTemplate,
+    "ui:options": {
+      label: false,
+    },
+    gscWithProductionOfUsefulEnergy: {
```
Comment (dleard) [inline bciers/apps/reporting/src/app/components/activities/uiSchemas/gscNonCompressionNonProcessingUiSchema.ts:36]:
```suggestion
    gscFuelOrWasteLinearFacilitiesUsefulEnergy: {
```

File: bciers/apps/reporting/src/app/components/activities/uiSchemas/gscNonCompressionNonProcessingUiSchema.ts [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+                  label: false,
+                  verticalBorder: true,
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+    gscWithoutProductionOfUsefulEnergy: {
```
Comment (dleard) [inline bciers/apps/reporting/src/app/components/activities/uiSchemas/gscNonCompressionNonProcessingUiSchema.ts:99]:
```suggestion
    General stationary combustion of fuel or waste at a linear facilities operation not resulting in the production of useful energy: {
```

File: bc_obps/reporting/migrations/0014_gsc_non_compression_non_processing.py
Comment (dleard) [inline bc_obps/reporting/migrations/0014_gsc_non_compression_non_processing.py:1]: All the source type names will need to be swapped in this file as above, should just need a quick search & replace

**Review by dleard — CHANGES_REQUESTED** [review]
Looks like this just needs a few source type name swaps. LFO GSC activities have similar sounding, but different source type names.
Tests are a nice addition!

**Review by dleard — APPROVED** [review]
Nice! I think we should add a tech-debt card to add tests to the activities we've already done, using these tests as a template & make sure that any new activity configurations are also tested

**Comment by pbastia** [comment]
Sounds good! I'll write a card.

**Pull request merged**